src/schema/v2/artwork/updateArtworkMutation.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  GraphQLString,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import Artwork from "schema/v2/artwork"
+import { omitBy } from "lodash"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+
+interface UpdateArtworkMutationInputProps {
+  id: string
+  availability?: boolean
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "updateArtworkSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    artwork: {
+      type: Artwork.type,
+      resolve: (artwork) => artwork,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "updateArtworkFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "updateArtworkResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const updateArtworkMutation = mutationWithClientMutationId<
+  UpdateArtworkMutationInputProps,
+  any,
+  ResolverContext
+>({
+  name: "UpdateArtworkMutation",
+  description: "Updates an artwork.",
+  inputFields: {
+    availability: {
+      type: GraphQLString,
+      description: "The availability of the artwork",
+    },
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The id of the artwork to update.",
+    },
+  },
+  outputFields: {
+    artworkOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: the artwork note updated.",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async ({ id, ...props }, { updateArtworkLoader }) => {
+    if (!updateArtworkLoader) {
+      return new Error("You need to be signed in to perform this action")
+    }
+
+    const payload = omitBy(props, (prop) => prop == null)
+
+    try {
+      const response = await updateArtworkLoader(id, payload)
+      return response
+    } catch (error) {
+      const formattedErr = formatGravityError(error)
```

Comment:
non-blocker: @mzikherman do we consider a helper for this kind of error? I've seen it in MP a few times.

Reply:
I've flagged this a lot too in various places 😆

Reply:
I'm not sure I understand, you mean this construction of try/catch? `formatGravityError` is itself already a helper...but I'm probably missing something.

Reply:
Yah meaning for all of these mutations we have to construct this try catch and so on. It could be a one-liner